controllers/movies.js

```diff
@@ -217,6 +217,7 @@ const renderFavoritesPageFiltered  = (req, res) => {
             let favoriteMovies = [];
             let favoriteCountArray = [];
             let favoritePromises = [];
+            let addedMovies = 0;
```

Comment:
Why did you need this variable here and not in the above method?

Reply:
I only need this variable when the results are actually filtered because when we are adding movies to the array to be passed to the ejs we are skipping over some of the movies.  In the other method, I knew that the array length would be equal to the number of movies returned but now I don't know the array length after the movies are filtered by genre.